### DISCUSSION TITLE
sleepwatcher: plist wrong if brew installed in /opt/homebrew

### DISCUSSION THREAD

**Discussion by drq883** [Homebrew/discussions#2679] [Everyday usage] [answered]
I was wondering if there was any way, in sleepwatcher.rb, to know the prefix where homebrew is installed and use that in the text, maybe:

   %brewprefix%/sbin/sleepwatcher

That way it will work no matter where homebrew installs it's Cellars, etc.

**Comment by gromgit** [comment, marked as answer]
The current `sleepwalker.rb` already contains:
```
  service do
    run [opt_sbin/"sleepwatcher", "-V", "-s", "#{ENV["HOME"]}/.sleep", "-w", "#{ENV["HOME"]}/.wakeup"]
    run_type :immediate
    keep_alive true
  end
```
which (as I understand it) should generate a plist that runs `${HOMEBREW_PREFIX}/opt/sleepwatcher/sbin/sleepwatcher`.

I don't have an M1 Mac to test, so if you're getting an incorrect plist generated, then:
```
brew update
brew reinstall sleepwatcher
brew services stop sleepwatcher
brew services start sleepwatcher
```
If that still doesn't fix things, you should probably file an issue.

**Reply by drq883** [reply]
I'm pretty sure I figured this one out. All works now with **brew install sleepwatcher** 

I had done a "migration assistant" of only my user data from my old MacBook backup to this new MacBook Air. I'm sure this brought over the old .plist and I guess brew saw that and didn't install a new one. I did a **brew uninstall sleepwatcher** then removed any lingering sleepwatcher files then reinstalled.

The new file **homebrew.mxcl.sleepwatcher.plist** contains:

```
<string>/opt/homebrew/opt/sleepwatcher/sbin/sleepwatcher</string>
```

Thank you for explaining the "service" do block in the .rb file. I was looking at the tarball only.
